### PR TITLE
Doc centralizing

### PR DESCRIPTION
**Pull request by AndrewDorn**
## Summary of Changes

Please write a sentence or two summarizing the proposed change.

Addresses [ORCA-73\](https://bugs.earthdata.nasa.gov/browse/ORCA-73)

## Changes

* Detailed list or prose of changes
* Centralized docs relating to testing and setup to the new Docusaurus site,

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the necessary documentation (remove those that do not apply)
    - [x] Testing documentation
    - [x] Development documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
    - [ ] Unit tests
    - [ ] Integration tests
    - [ ] Manual tests (outlined above)
- [ ] New and existing unit tests pass locally with my changes
    - [ ] Automated tests passing (if not fork)
    - [ ] Manual testing/integration testing passes (if forked)
- [x] Any dependent changes have been merged and published in downstream modules
- [x] My code has passed security scanning
    - [x] Snyk
    - [x] git-secrets

### CODE REVIEW

File: tasks/copy_files_to_archive/README.md [hunk 1]
```diff
@@ -2,98 +2,10 @@
...
 
-- [Setup](#setup)
-- [Development](#development)
-  * [Unit Testing and Coverage](#unit-testing-and-coverage)
-  * [Linting](#linting)
 - [Deployment](#deployment)
   * [Deployment Validation](#deployment-validation)
 - [pydoc copy_files_to_archive](#pydoc-copy-files)
 
-<a name="setup"></a>
```
Comment (ifestus) [inline tasks/copy_files_to_archive/README.md:13]: I really like this change - moving things to a centralized location is a good idea.

**Review by ifestus — COMMENTED** [review]
I really like these changes. One question I have is now that the linting/testing information isn't in each task directory, are we expecting developers to look at the website/docs/ location (or the online documentation) to figure out how to build/test? I can see the value in that, but I can also see how that might be confusing for new developers.

Maybe a good in-between would be to keep the changes that you've made and then note in a README somewhere that the documentation regarding testing/building can be found at location `x`?

File: website/README.md [hunk 2]
```diff
@@ -1,5 +1,5 @@
 # Website
-
+# todo: Too many readmes and methods of starting. Choose one.
```
Comment (AndrewDorn) [inline website/README.md:2]: The instructions here conflict with the instructions in the base README.md

**Comment by AndrewDorn** [comment]
> I really like these changes. One question I have is now that the linting/testing information isn't in each task directory, are we expecting developers to look at the website/docs/ location (or the online documentation) to figure out how to build/test? I can see the value in that, but I can also see how that might be confusing for new developers.
> 
> Maybe a good in-between would be to keep the changes that you've made and then note in a README somewhere that the documentation regarding testing/building can be found at location `x`?

We had something to that effect, but it was at the bottom of the file. I've now moved it to the top, since we want to prioritize the new docusaurus pages.

File: website/docs/developer/development-guide/code/linting.md [hunk 3]
```diff
@@ -0,0 +1,13 @@
+---
+id: linting
+title: Running PyLint Against Your Code
+description: Instructions on running PyLint.
+---
+
+## Running PyLint
+1. Navigate to the task's base folder.
+1. Activate the virtual environment.
```
Comment (bhazuka) [inline website/docs/developer/development-guide/code/linting.md:9]: May want to add a step or reference a link to setting up the virtual environment.

File: website/docs/developer/development-guide/code/linting.md [hunk 4]
```diff
@@ -0,0 +1,13 @@
...
+1. Run pylint and fix any discovered issues if you can.
```
Comment (bhazuka) [inline website/docs/developer/development-guide/code/linting.md:10]: May want to add some comments about creating/maintaining a .pylintrc file to handle false positives. Possibly add some reference links for pylint.

File: website/docs/developer/development-guide/code/setup-dev-env.md [hunk 5]
```diff
@@ -0,0 +1,61 @@
...
+            venv\Scripts\deactivate.bat
+            ```
+   1. Linux/[Cmder](https://cmder.net/):  
+      Create:
+            ```
+            python -m venv venv
+            ```  
+      Activate:
+            ```
+            source venv\Scripts\activate
```
Comment (bhazuka) [inline website/docs/developer/development-guide/code/setup-dev-env.md:40]: This path should be `venv/bin/activate` We may be able to simplify this by pointing to the Python documentation too.  See https://docs.python.org/3.8/library/venv.html

File: website/docs/developer/development-guide/code/setup-dev-env.md [hunk 6]
```diff
@@ -0,0 +1,61 @@
...
+   Make sure Python is added to your PATH.
+1. Install the AWS client. View the [official documentation](https://docs.aws.amazon.com/cli/latest/userguide/cli-chap-welcome.html) if needed.
+    ```commandline
+    pip3 install awscli
+    ```
+
+## Per-Lambda Setup
+Lambdas should be built and tested within individual Virtual Environments.
+1. Navigate to the task's folder within 'culumus-orca\tasks\[task name]'.
+1. Create a new Virtual Environment and enter it.
```
Comment (bhazuka) [inline website/docs/developer/development-guide/code/setup-dev-env.md:19]: Possibly change from `Create a new Virtual Environment and enter it.` to `Create and activate a new Python virtual environment.`

File: website/docs/developer/development-guide/code/setup-dev-env.md [hunk 7]
```diff
@@ -0,0 +1,61 @@
...
+            ```
+1. With the VEnv activated, install requirements.
+   ```commandline
+   pip install boto3
+   pip install -r requirements.txt
+   ```
+   If requirements.txt references db_dbutils and/or pg_utils, install them with their respective commands.
+   ```commandline
+   pip install ..\dr_dbutils\dist\dr_dbutils-1.0.tar.gz
+   pip install ..\pg_utils\dist\pg_utils-1.0.tar.gz
```
Comment (bhazuka) [inline website/docs/developer/development-guide/code/setup-dev-env.md:54]: We should probably review this and create a card to work on later where we remove a distributed tar out of our repository. Ideally, these libraries should either be loaded to someplace like PyPI or referenced as part of our distribution.

File: website/docs/developer/development-guide/code/setup-dev-env.md [hunk 8]
```diff
@@ -0,0 +1,61 @@
...
+   ```
+1. If you are going to develop code, install development requirements.
```
Comment (bhazuka) [inline website/docs/developer/development-guide/code/setup-dev-env.md:56]: Remove `If you are going to develop code,` lets assume they are developing code if they are reading this and performing the actions.

File: website/docs/developer/development-guide/code/setup-dev-env.md [hunk 9]
```diff
@@ -0,0 +1,61 @@
...
+   ```commandline
+   pip install coverage
+   pip install pylint
+   pip install -r requirements-dev.txt
+   ```
```
Comment (bhazuka) [inline website/docs/developer/development-guide/code/setup-dev-env.md:61]: May want to add another step to setup the document environment and point to that document.

File: website/docs/developer/development-guide/code/unit-tests.md [hunk 10]
```diff
@@ -0,0 +1,17 @@
...
+1. Navigate to the task's base folder.
+1. Activate the virtual environment.
+1. Run
+    ```commandline
+    coverage run --source [name of lambda] -m pytest
+    ```
+1. Output the coverage results to the file system by running
+    ```commandline
+    coverage html
+    ```
```
Comment (bhazuka) [inline website/docs/developer/development-guide/code/unit-tests.md:17]: Maybe put some comments in that we expect at least 80% coverage from uint tests and that all unit-test pass. Also, all new code must have a unit test.

You may also want to put some notes in about the test harness we are using `pytest` and  expectations with some links. I would also add any code/styling standards for the tests we may want to ask for/enforce for the PR.

File: website/docs/developer/development-guide/code/postgres-tests.md [hunk 11]
```diff
@@ -0,0 +1,30 @@
...
+description: Instructions on running 'postgres' tests.
+---
+
+## Preparing to Run Postgres Tests
+Test files ending in '_postgres.py' require a postgres database to be accessible.
+
+1. Make sure you have Docker running on your machine.
+1. Open a command prompt and run
+    ```commandline
+    docker run -it --rm --name some-postgres -v [path to repository]/database/ddl/base:/docker-entrypoint-initdb.d/ -p 5432:5432 -e POSTGRES_PASSWORD=postgres postgres
```
Comment (bhazuka) [inline website/docs/developer/development-guide/code/postgres-tests.md:13]: should change out the `POSTGRES_PASSWORD=postgres` with an environment variable like so: `POSTGRES_PASSWORD=$PG_PASSWORD` or something similar.

**Review by bhazuka — CHANGES_REQUESTED** [review]
I agree with Jacob's comment that we should probably link to our documentation on environment setup and testing as part of our task templates.

File: website/README.md (hunk 2 above)
[the remark this answers is not in the document]
Comment (AndrewDorn) [reply website/README.md:2]: I was hoping you'd say that. Done.

File: website/docs/developer/development-guide/code/code-intro.md [hunk 12]
```diff
@@ -5,3 +5,5 @@ desc: Provides basic information on ORCA code including layout, standards and ot
 ---
 
 Content for this page is being created.
+
+Run through the steps in [Setting Up a Dev Environment](setup-dev-env) prior to modifying/testing code.
```
[the remark this answers is not in the document]
Comment (AndrewDorn) [reply website/docs/developer/development-guide/code/code-intro.md:9]: Double-checked, and it does indeed work.

File: website/docs/developer/development-guide/code/setup-dev-env.md [hunk 13]
```diff
@@ -0,0 +1,61 @@
+---
+id: setup-dev-env
+title: Setting Up a Dev Environment
+description: Instructions on creating an environment for working with lambdas.
+---
+
```
[the remark this answers is not in the document]
Comment (AndrewDorn) [reply website/docs/developer/development-guide/code/setup-dev-env.md:6]: Probably. I was focusing on a no-frills dev setup for the time being, but I'd definitely be in favor of more installation stuff. Shall I create a card.

File: website/docs/about/introduction/glossary.md [hunk 14]
```diff
@@ -199,7 +199,9 @@ used to create Workflows.
 ## U
 
 ## V
-
+###Virtual Environment (VEenv)
+A contained environment for building/running Python code.
+See [Setting Up a Dev Environment](../../developer/testing/setup-dev-env) for more information.
```
[the remark this answers is not in the document]
Comment (AndrewDorn) [reply website/docs/about/introduction/glossary.md:204]: And my title was formatted incorrectly. Thanks.

File: website/docs/about/introduction/glossary.md [hunk 15]
```diff
@@ -199,7 +199,9 @@ used to create Workflows.
 ## U
 
 ## V
-
+###Virtual Environment (VEenv)
```
[the remark this answers is not in the document]
Comment (AndrewDorn) [reply website/docs/about/introduction/glossary.md:202]: You're right, that's the official name. Fixed.

File: website/docs/developer/development-guide/code/linting.md (hunk 3 above)
Comment (AndrewDorn) [reply website/docs/developer/development-guide/code/linting.md:9]: I was hoping they'd read the intro first, but you're right. Should I put a 'tip' on each affected page then?

File: website/docs/developer/development-guide/code/unit-tests.md (hunk 10 above)
Comment (AndrewDorn) [reply website/docs/developer/development-guide/code/unit-tests.md:17]: Updated command, and added link.

File: website/docs/developer/development-guide/code/postgres-tests.md (hunk 11 above)
Comment (AndrewDorn) [reply website/docs/developer/development-guide/code/postgres-tests.md:13]: This step occurs before these environment variables exist.

File: website/docs/developer/development-guide/code/setup-dev-env.md (hunk 5 above)
Comment (AndrewDorn) [reply website/docs/developer/development-guide/code/setup-dev-env.md:40]: Updated command, and added link.

File: website/docs/developer/development-guide/code/setup-dev-env.md (hunk 6 above)
Comment (AndrewDorn) [reply website/docs/developer/development-guide/code/setup-dev-env.md:19]: Updated.

File: website/docs/developer/development-guide/code/setup-dev-env.md (hunk 7 above)
Comment (AndrewDorn) [reply website/docs/developer/development-guide/code/setup-dev-env.md:54]: Agreed.

File: website/docs/developer/development-guide/code/setup-dev-env.md (hunk 8 above)
Comment (AndrewDorn) [reply website/docs/developer/development-guide/code/setup-dev-env.md:56]: True. Fixed.

File: website/docs/developer/development-guide/code/setup-dev-env.md (hunk 9 above)
Comment (AndrewDorn) [reply website/docs/developer/development-guide/code/setup-dev-env.md:61]: As in, this website?

File: website/docs/developer/development-guide/code/unit-tests.md (hunk 10 above)
Comment (AndrewDorn) [reply website/docs/developer/development-guide/code/unit-tests.md:17]: Sounds good. I'll get started on a unit-test dev guide section.

File: website/docs/developer/development-guide/code/setup-dev-env.md (hunk 13 above)
[the remark this answers is not in the document]
Comment (AndrewDorn) [reply website/docs/developer/development-guide/code/setup-dev-env.md:6]: Card created.

File: website/docs/developer/development-guide/code/linting.md (hunk 3 above)
Comment (AndrewDorn) [reply website/docs/developer/development-guide/code/linting.md:9]: Done

File: website/docs/developer/development-guide/code/linting.md (hunk 4 above)
Comment (AndrewDorn) [reply website/docs/developer/development-guide/code/linting.md:10]: Task Created

File: website/docs/developer/development-guide/code/postgres-tests.md (hunk 11 above)
Comment (AndrewDorn) [reply website/docs/developer/development-guide/code/postgres-tests.md:13]: Done

File: website/docs/developer/development-guide/code/setup-dev-env.md (hunk 9 above)
Comment (AndrewDorn) [reply website/docs/developer/development-guide/code/setup-dev-env.md:61]: Added

File: website/docs/developer/development-guide/code/unit-tests.md (hunk 10 above)
Comment (AndrewDorn) [reply website/docs/developer/development-guide/code/unit-tests.md]: Added something

File: website/docs/developer/development-guide/code/code-intro.md [hunk 16]
```diff
@@ -5,3 +5,5 @@ desc: Provides basic information on ORCA code including layout, standards and ot
 ---
 
 Content for this page is being created.
```
[the remark this answers is not in the document]
Comment (AndrewDorn) [reply website/docs/developer/development-guide/code/code-intro.md:7]: https://bugs.earthdata.nasa.gov/browse/ORCA-124

**Review by bhazuka — CHANGES_REQUESTED** [review]

**Review by bhazuka — CHANGES_REQUESTED** [review]
In the tasks change the link [Docusaurus Pages](../../website/docs/developer/development-guide/code/code-intro.md) to [developer guide]((../../website/docs/developer/development-guide/code/code-intro.md)

**Review by bhazuka — APPROVED** [review]

**Pull request merged**